wrapper/src/main/java/software/amazon/jdbc/exceptions/AbstractPgExceptionHandler.java

```diff
@@ -62,8 +62,14 @@ public boolean isLoginException(final Throwable throwable) {
       if (exception instanceof SQLLoginException) {
         return true;
       }
+
+      String sqlState = null;
       if (exception instanceof SQLException) {
-        return isLoginException(((SQLException) exception).getSQLState());
+        sqlState = ((SQLException) exception).getSQLState();
+      }
+
+      if (isLoginException(sqlState)) {
+        return true;
```

Comment:
Super tiny nitpick: 
I think the above could be simplified like this. Not sure if its less readable now though due to the longer condition.
```
      if (exception instanceof SQLException 
          && isLoginException(((SQLException) exception).getSQLState())) {
        return true;
      }
```

What do you think?

Reply:
Yeah I wasn't sure which of these two was preferable, I'm fine with either one

Reply:
Yeah, same here. Im fine with either one. 
I'll slap an approval on it :P